### PR TITLE
fflogs: init at 8.3.42

### PR DESCRIPTION
**Pull request by keysmashes**
## Description of changes

Add package `fflogs`, an application to upload combat logs to fflogs.com ([help page](https://articles.fflogs.com/help/uploader)).

There's a macOS version, but I'm not able to test/maintain it.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/fflogs/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, appimageTools
+, fetchurl
+}:
+
+let
+  pname = "fflogs";
+  version = "7.0.124";
```
Comment (FliegendeWurst) [inline pkgs/games/fflogs/default.nix:8]: 7.0.127 has been released by now.

File: pkgs/games/fflogs/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  '';
+
+  meta = with lib; {
+    description = "An application for uploading Final Fantasy XIV combat logs to fflogs.com";
+    homepage = "https://www.fflogs.com/client/download";
+    downloadPage = "https://github.com/RPGLogs/Uploaders-fflogs/releases/latest";
+    license = licenses.unfree; # no license listed
+    mainProgram = "fflogs-${version}";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ sersorrel ];
```
Comment (FliegendeWurst) [inline pkgs/games/fflogs/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ sersorrel ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]
Builds successfully on x64 Linux.

**Comment by sersorrel** [comment]
Rebased, now #271071 is merged.

File: pkgs/by-name/ff/fflogs/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+, appimageTools
+, fetchurl
+}:
+
+let
+  pname = "fflogs";
+  version = "8.3.42";
+  src = fetchurl {
+    url = "https://github.com/RPGLogs/Uploaders-fflogs/releases/download/v${version}/fflogs-v${version}.AppImage";
+    sha256 = "08mkixjpppijssmg63l542xhmxyr7xs9z1v9qfjda7cfqgah5kh5";
```
Comment (pbsds) [inline pkgs/by-name/ff/fflogs/package.nix:11]: please use the SRI hash format

```suggestion
    hash = "sha256-...";
```

File: pkgs/by-name/ff/fflogs/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  extracted = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications
+    cp -r ${extracted}/usr/share/icons $out/share/
+    cp ${extracted}/fflogs.desktop $out/share/applications/
+    sed -i s@^Exec=.\*@Exec=$out/bin/fflogs@g $out/share/applications/fflogs.desktop
```
Comment (pbsds) [inline pkgs/by-name/ff/fflogs/package.nix:22]: please avoid nix store paths in desktop entries

```suggestion
    sed -i s@^Exec=.\*@Exec=fflogs@g $out/share/applications/fflogs.desktop
```

File: pkgs/by-name/ff/fflogs/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  '';
+
+  meta = with lib; {
+    description = "An application for uploading Final Fantasy XIV combat logs to fflogs.com";
+    homepage = "https://www.fflogs.com/client/download";
+    downloadPage = "https://github.com/RPGLogs/Uploaders-fflogs/releases/latest";
+    license = licenses.unfree; # no license listed
+    mainProgram = "fflogs-${version}";
```
Comment (pbsds) [inline pkgs/by-name/ff/fflogs/package.nix:30]:
```suggestion
    mainProgram = "fflogs";
```

File: pkgs/by-name/ff/fflogs/package.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "08mkixjpppijssmg63l542xhmxyr7xs9z1v9qfjda7cfqgah5kh5";
+  };
+  extracted = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications
+    cp -r ${extracted}/usr/share/icons $out/share/
```
Comment (pbsds) [inline pkgs/by-name/ff/fflogs/package.nix:20]: please move `share/icons/hicolor/0x0/apps/fflogs.png` to `share/icons/hicolor/512x512/apps/fflogs.png`

File: pkgs/by-name/ff/fflogs/package.nix (hunk 4 above)
Comment (sersorrel) [reply pkgs/by-name/ff/fflogs/package.nix]: done, though I'm curious about the reasoning

Comment (pbsds) [reply pkgs/by-name/ff/fflogs/package.nix]: some desktop environments like gnome will copy dekstop entries to `~/.config/autostart` to schedule them for automatic start on login. Store paths may go out of date and get garbage collected, causing the startup entries to stop working.

**Comment by pbsds** [comment]
```
fflogs> mv: cannot stat '/nix/store/sfl8zyc5pxw6bbgk14dnqhapg9rqxkq0-fflogs-8.3.42/share/icons/0x0': No such file or directory
```

**Comment by sersorrel** [comment]
Thank *you* for merging this!

**Pull request merged**